src/main/java/de/dhbw/karlsruhe/services/MenuDialogService.java

```diff
@@ -0,0 +1,72 @@
+package de.dhbw.karlsruhe.services;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class MenuDialogService {
+    private Scanner scanner;
+    private String[] menuOptions;
+    private int userInput;
+    private PlayDialogService playDialogService;
+    private LeaderboardDialogService leaderboardDialogService;
+    private LogoutService logoutService;
+
+    public MenuDialogService() {
+        this.scanner = new Scanner(System.in);
+        this.menuOptions = new String[]{"Play", "Leaderboard", "Logout"};
+    }
+
+    public void startMenuDialog() {
+        this.displayMenuOptions();
+        this.userInput = this.awaitUserInput();
+        this.checkUserInput(this.userInput);
+    }
```

Comment:
Vielleicht sollte das ganze bis Logout gewählt wird in einer Schleife stattfinden, damit nach zum Beispiel einem fertigen Sudoku wieder das Hauptmenu kommt .

Reply:
Sehr guter Hinweis, danke!
Würde ich implementieren, sobald die Logout Funktionalität reingemerged wurde.